lms/js_config_types.py

```diff
@@ -0,0 +1,27 @@
+from typing import NotRequired, TypedDict
+
+
+class APICallInfo(TypedDict):
+    path: str
+    authUrl: NotRequired[str]
+
+
+class APIAssignment(TypedDict):
+    id: int
+    title: str
+
+
+class APIAssignmentStats(TypedDict):
+    display_name: str
+    annotations: int
+    replies: int
+    last_activity: str
+
+
+class DashboardLinks(TypedDict):
+    assignmentApi: str
+    assignmentStatsApi: str
```

Comment:
I would be inclined to organize everything that is an API route, as opposed to a link, under an appropriate top-level struct. Then you don't need to add an "Api" suffix to everything.

Reply:
Yep. Something like:

```json
{
  "routes": {
    "assignmentDetails": {...},
    "assignmentStats": {...},
    ...
  }
}
```

Reply:
Gone for that approach